### ISSUE TITLE
Type only imports are not set as dependencies

### ISSUE THREAD

**Issue by ezolenko** [ezolenko/rollup-plugin-typescript2#7]
After interface modification the plugin (v0.3.0) serves depending files from the cache, missing potentially breaking changes.

To reproduce make a clean build, break a type-only ts file and rerun a build.

**Comment by ezolenko** [comment]
Might depend on resolution of one of those:
https://github.com/rollup/rollup/issues/1203
https://github.com/rollup/rollup/issues/1248

**Comment by ksnyde** [comment]
Can't wait to see this included, looks like your blockers are starting to have PR's against them.

**Comment by ezolenko** [comment]
Also affects watch mode, need to add all files specified by tsconfig to rollup watch list

**Comment by posva** [comment]
is this error the same:

```
(!) Import of non-existent export
src/rtdb/utils.ts
1: import { database } from 'firebase';
```

`database` is type only and there isn't a real export in the js file:

```
(!) Error when using sourcemap for reporting an error: Can't resolve original location of error.
src/rtdb/utils.ts: (1:9)
[!] Error: 'database' is not exported by ../../../node_modules/firebase/dist/index.esm.js
```

**Comment by ezolenko** [comment]
No, your error looks different. You can't really import anything a module is not exporting. If you want side effects, you can do plain `import 'firebase'`. What happens if you simply remove that import?

**Comment by posva** [comment]
If I remove it it doesn't compile because it's missing that `database` namespace. It's weird because the error only happens for `packages/@posva/vuefire-core` but not for `packages/vuefire` (in a learn monorepo). I ended up creating an alias (rollup-plugin-alias) for the `firebase` package that resolves to a stub that exports empty objects. It suppress the errors with rollup while still getting correct typings output:

**Comment by domoritz** [comment]
I think this bug causes a problem in my builds as well. I am missing declaration files for files that export only types. Is this captured by this issue? Are there any known workarounds?

**Comment by agilgur5** [comment]
Wrote up a root cause analysis for this in https://github.com/ezolenko/rollup-plugin-typescript2/issues/298#issuecomment-1146658442.

Also wrote up a PR that is a _partial_ fix for this in #345

**Comment by agilgur5** [comment]
This has been fixed in #406 and released in [`0.34.0`](https://github.com/ezolenko/rollup-plugin-typescript2/releases/tag/0.34.0) 🎉
